include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1033,58 @@ namespace GridTools
   ( const MeshType                                                                    &mesh,
     const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate );
 
+  /**
+   * Given a cell @p parent_cell of a mesh a predicate and a flag @p has_predicate computes
+   * the bounding box of all descendants cells of @p parent_cell (or the cell itself, if
+   * it is active) which have the predicate property.
+   *
+   * This is an example of the predicate which is needed to find the bounding box of
+   * locally owned cells:
+   * @code
+   * template <int dim>
+   * bool
+   * pred_locally_owned(const typename Triangulation<dim>::active_cell_iterator &cell)
+   * {
+   *   return cell->is_locally_owned();
+   * }
+   * @endcode
+   *
+   * This function doesn't work with curved meshes.
+   */
+  template < class MeshType >
+  BoundingBox < MeshType::space_dimension >
+  compute_cell_predicate_bounding_box
+  (const typename MeshType::cell_iterator &parent_cell,
+   const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate,
+   bool &has_predicate);
+
+  /**
+   * Compute a vector of bounding boxes which encloses the space occupied by
+   * the cells which have the predicate property
+   *
+   * The algorithm loops over all cells at given level @p refinement_level and active cells at lower refinement
+   * levels, creating a Bounding Box for each of them. Then it merges boxes which can be expressed as a single one.
+   *
+   * The parameters to control the algorithm are:
+   * - @p predicate : the property of the cells to enclose (see compute_cell_predicate_bounding_box)
+   * - @p refinement_level : it describes the level at which the initial Bounding Boxes are created. The refinement
+   *  should be set to a low refinement level. If @p refinement_level is higher than the number of levels of the
+   *  triangulation or then an exception is trown.
+   * - @p max_boxes : the maximum number of bounding boxes to compute. If more are created the smaller ones are
+   *  merged with neighbours. If the value is set to 0 all computed bounding boxes are returned.
+   *  Notice only neighbouring cells are merged (see the @p is_neighbour function in bounding box class): if
+   *  the target number of bounding boxes max_boxes can't be reached by merging neighbours an exception is trown
+   * - @p err (optional) : this value is use to cope with numerical errors. Coordinates which are closer than
+   *  this value are considered the same
+   *
+   * Notice this function doesn't work with curved meshes.
+   */
+  template < class MeshType >
+  std::vector< BoundingBox< MeshType::space_dimension > >
+  compute_mesh_predicate_bounding_box
+  ( const MeshType &mesh,
+    const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate,
+    const unsigned int &refinement_level, const unsigned int &max_boxes, const double &err = 1e-12);
```

Comment:
you can set default to `refinement_level` being the coarsest level in `mesh`.
For `max_boxes` set to `numbers::invalid_unsigned_int` (and use as default) to return all boxes. Add a check that the number is positive.
Finally, you don't need `err` here either